modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c

```diff
@@ -11,10 +11,12 @@
 
 #include <compiler_abstraction.h>
 #include <zephyr/kernel.h>
-#if defined(CONFIG_CLOCK_CONTROL_NRF)
 #include <zephyr/drivers/clock_control/nrf_clock_control.h>
```

Comment:
You could get rid of these conditional directives. `nrf_clock_control.h` is all you actually need to include here.